kymatio/scattering1d/backend/torch_backend.py

```diff
@@ -56,59 +56,19 @@ def pad_1d(x, pad_left, pad_right, mode='constant', value=0.):
     pad_right : int
         amount to add on the right of the tensor (at the end of the temporal
         axis).
-    mode : string, optional
-        Padding mode. Options include 'constant' and 'reflect'. See the
-        PyTorch API for other options.  Defaults to 'constant'.
-    value : float, optional
-        If mode == 'constant', value to input within the padding. Defaults to
-        0.
-
     Returns
     -------
     res : tensor
         The tensor passed along the third dimension.
     """
     if (pad_left >= x.shape[-1]) or (pad_right >= x.shape[-1]):
-        if mode == 'reflect':
-            raise ValueError('Indefinite padding size (larger than tensor).')
-    res = F.pad(x.unsqueeze(2),
+        raise ValueError('Indefinite padding size (larger than tensor).')
+    res = F.pad(x[:, :, None],
                 (pad_left, pad_right, 0, 0),
-                mode=mode, value=value).squeeze(2)
+                mode='reflect')
+    res = res[:, :, 0, ..., None]
```

Comment:
My take is that: Incoming `x` shape is `(batch, channel, time)`. 
The padding operator was designed for 2D images, so we need something of shape `batch, channel, 1, time` and then `pad_left, pad_right` pad on the last axis, and `0, 0` pad on the penultimate axis.

Now, I have the feeling that torch padding has since developed into something more general, making this whole procedure obsolete. @MuawizChaudhary do you mind checking that? If you can pad without adding axes that would simplify the code. If not, then this is good to go.

Reply:
I don't believe I see anything from the 1.4.0 patch notes or the 1.5.0 patch notes. I'll try padding without all the adding and removing dimensions business.

Reply:
Seems like you're right!